ClassicLibrary.c

```diff
@@ -4,6 +4,68 @@
 // Ayisha Sameera,GET.
 // ------------------------------------------------------------------
 // ClassicLibrary.c
-// Program on branch main.
+// Program on branch A5.
 // Program that contains the definiton of functions defined in Header.h.
-// ------------------------------------------------------------------------------------------------
\ No newline at end of file
+// ------------------------------------------------------------------------------------------------
+
+// Heap Sort inserts elements from an unsorted array into a max heap.A Max Heap is a complete 
+// binary tree,where value of each node is greater than or equal to the values of its children. 
+// The same property must be recursively true for all nodes in the binary tree i.e Max Heap stores 
+// the maximum value at the root of the heap. An array is first constructed to form a max heap, 
+// then the root of the max heap is swapped with the last element of the unsorted array recursively.
+// This swap breaks the max heap property so we "heapify" it again. 
+// We continue until heap size becomes one.
+
+
+/// <summary>
+/// Swaps the largest element (root of max heap) with the last element of the unsorted array
+/// </summary>
+void Swap (int* a, int* b) {
+   if (*a != *b) {

```

Comment:
```suggestion
   if (*a == *b) return;

```
See [Guard](https://en.wikipedia.org/wiki/Guard_(computer_science))